geom/type_envelope.go

```diff
@@ -7,28 +7,30 @@ import (
 	"github.com/peterstace/simplefeatures/rtree"
 )
 
-// Envelope is an axis-aligned rectangle (also known as an Axis Aligned
-// Bounding Box or Minimum Bounding Rectangle). It usually represents a 2D area
-// with non-zero width and height, but can also represent degenerate cases
-// where the width or height (or both) are zero. Its bounds are validated so as
-// to not be NaN or +/- Infinity.
+// Envelope is a generalised axis-aligned rectangle (also known as an Axis
+// Aligned Bounding Box or Minimum Bounding Rectangle). It usually represents a
+// 2D area with non-zero width and height. But it can also represent degenerate
+// cases where the width or height (or both) are zero, or the envelope is
+// empty. Its bounds are validated so as to not be NaN or +/- Infinity.
+//
+// An envelope can be thought of as as being similar to a regular geometry, but
+// can only represent an empty geometry, a single point, a horizontal or
+// vertical line, or an axis aligned rectangle with some area.
+//
+// The Envelope zero value is the empty envelope. Envelopes are immutable after
+// creation.
 type Envelope struct {
-	min XY
-	max XY
+	full bool
```

Comment:
Curious, what were the reasons for choosing `full` instead of `empty`?

Reply:
It's mainly to do with what the `Envelope` zero value ends up being. All of the zero values of the other types represent the "empty" variant, so I think it's a good idea to do that here too for consistency. If we were to use `empty` instead of `full` as the field, then the zero value would be the single-point envelope at `(0, 0)`.

Reply:
I wonder if "full" is the wrong word here to best describe the state though?

The _best_ description of what the field really means is "non-empty", although IMO any logic involving boolean negations when the name of the boolean itself contains a negation can be pretty confusing...

Reply:
> All of the zero values of the other types represent the "empty" variant, so I think it's a good idea to do that here too for consistency.

Ah, this is a great reason to use `full` and I think is meaningful enough to easily understand its intent; although it very subtly implies that there are three states, instead of a binary value: 
- full
- not full:
  - empty
  - partially full

> I wonder if "full" is the wrong word here to best describe the state though?

Just brainstorming some other words if you're still looking to rename it: `set`, `isSet`, `initialised`, `isInitialised` (or its American variant), `populated`, `isPopulated`.

Or another alternative is to define an enum of these two values that makes it clearer:
```
const (
   full = iota
   empty
)
```